### PR TITLE
Small MC refactor

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
- Splits the MC tab into multiple tabs (see below images)
  - We had so many subsystems that it was getting ridiculous and I couldnt actually open the tab on prod, which is an issue being the host
- Removes some subsystems and functionality we will never use (Lavaland medals for example)
- Converts `SSalarm`, `SSipintel` and `SSdiscord` to global datums since they dont initialize or fire, and if an SS doesnt do either of those, it shouldnt be an SS (CI lint added for this)
- SS initialize was changed to behave more like atom initialize, where you call the parent first instead of last
- Added a toggle verb to show/hide all the MC tabs
- Removed some fire priority and init orders we will never use

## Why It's Good For The Game
See all of my above reasoning

## Images of changes
<details>
<summary>New tabs</summary>
Metadata

image

Offline SSs

image

Low intensity SSs (note icon smoothing is offline here, it flips between on and off based on queue load)

image

Default intensity (note some of these are low, I just dont want to bloat the low tab)

image

High intensity

image
</details>

## Testing
I ensured the SSs still initialized. Probably worth a TM.

## Changelog
:cl: 
add: [Admin] Split the MC tab to multiple tabs, as well as a preference to show/hide these tabs
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
GitHub if you open one more of my PRs as a draft, there will be words.

File: code/modules/client/preference/preferences_toggles.dm [hunk 1]
```diff
@@ -357,3 +357,11 @@
...
 	to_chat(src, "Your key will [(client.prefs.toggles2 & PREFTOGGLE_2_ANON) ? "no longer" : "now"] be shown in certain events (end round reports, deadchat, etc).</span>")
 	client.prefs.save_preferences(src)
+
+/client/proc/toggle_mctabs()
+	set name = "Show/Hide MC Tabs"
+	set category = "Preferences"
+	set desc = "Shows or hides the MC tabs."
+	prefs.toggles2 ^= PREFTOGGLE_2_MC_TABS
+	prefs.save_preferences(src)
+	to_chat(src, "You will [(prefs.toggles2 & PREFTOGGLE_2_MC_TABS) ? "no longer" : "now"] see the MC tabs on the top right.")
```
Comment (AffectedArc07) [inline code/modules/client/preference/preferences_toggles.dm:367]: I need to fix this

**Comment by S34NW** [comment]
These better start closed or I'm going to scream

**Comment by AffectedArc07** [comment]
> These better start closed or I'm going to scream

The tab pref starts disabled yes.

**Comment by AffectedArc07** [comment]
I cant type. Above commit is meant to say it nabs https://github.com/tgstation/tgstation/pull/72324

File: code/__DEFINES/preferences.dm [hunk 2]
```diff
@@ -59,8 +59,9 @@
 #define PREFTOGGLE_2_SEE_ITEM_OUTLINES 	4096
 #define PREFTOGGLE_2_HIDE_ITEM_TOOLTIPS  8192
 #define PREFTOGGLE_2_THOUGHT_BUBBLE	16384
+#define PREFTOGGLE_2_MC_TABS	32768
```
Comment (Sirryan2002) [inline code/__DEFINES/preferences.dm:62]: shouldn't these be converted to bit shifts? 
i.e
```dm
#define PREFTOGGLE_2_MC_TABS	(1 << 15)

File: code/controllers/subsystem/tickets/mentor_tickets.dm [hunk 3]
```diff
@@ -21,10 +21,10 @@ GLOBAL_REAL(SSmentor_tickets, /datum/controller/subsystem/tickets/mentor_tickets
 	db_save_id = "MENTOR"
 
 /datum/controller/subsystem/tickets/mentor_tickets/Initialize()
+	..()
```
Comment (Sirryan2002) [inline code/controllers/subsystem/tickets/mentor_tickets.dm:24]: why bother calling parent here if you're just going to override `close_messages` ?

File: code/datums/alarm_manager.dm [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+/datum/alarm_manager
+	var/list/alarms = list(
+		"Motion" = list(),
+		"Fire" = list(),
+		"Atmosphere" = list(),
+		"Power" = list(),
+		"Burglar" = list()
+	)
+
+/datum/alarm_manager/proc/triggerAlarm(class, area/A, list/O, obj/alarmsource)
```
Comment (Sirryan2002) [inline code/datums/alarm_manager.dm:12]: is there any reason that these don't use snake_case ?

File: code/game/machinery/computer/station_alert.dm [hunk 5]
```diff
@@ -42,12 +42,12 @@
...
 	data["alarms"] = list()
-	for(var/class in SSalarm.alarms)
+	for(var/class in GLOB.alarm_manager.alarms)
 		if(!(class in alarms_listend_for))
 			continue
 		data["alarms"][class] = list()
-		for(var/area in SSalarm.alarms[class])
-			for(var/thing in SSalarm.alarms[class][area][3])
+		for(var/area in GLOB.alarm_manager.alarms[class])
+			for(var/thing in GLOB.alarm_manager.alarms[class][area][3])
```
Comment (Sirryan2002) [inline code/game/machinery/computer/station_alert.dm:50]: the listception here feels like a crime, I love it

File: code/modules/mob/mob.dm [hunk 6]
```diff
@@ -978,16 +978,19 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
 			add_spell_to_statpanel(S)
 
 	// Allow admins + PR reviewers to VIEW the panel. Doesnt mean they can click things.
-	if(is_admin(src) || check_rights(R_VIEWRUNTIMES, FALSE))
-		if(statpanel("MC")) //looking at that panel
+	if((is_admin(src) || check_rights(R_VIEWRUNTIMES, FALSE)) && client?.prefs.toggles2 & PREFTOGGLE_2_MC_TABS)
```
Comment (Sirryan2002) [inline code/modules/mob/mob.dm:981]:
```suggestion
	if((is_admin(src) || check_rights(R_VIEWRUNTIMES, FALSE)) && (client?.prefs.toggles2 & PREFTOGGLE_2_MC_TABS))
```
bitwise styling

File: code/modules/mob/mob.dm [hunk 7]
```diff
@@ -996,10 +999,38 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
...
+
+		// Shows subsystems with the SS_CPUDISPLAY_HIGH flag
+		if(statpanel("MC|H"))
+			stat("Info", "Showing subsystems marked as high intensity")
 			if(Master)
-				stat(null)
-				for(var/datum/controller/subsystem/SS in Master.subsystems)
-					SS.stat_entry()
+				for(var/datum/controller/subsystem/SS as anything in Master.subsystems)
+					if((SS.cpu_display == SS_CPUDISPLAY_HIGH) && !(SS.flags & SS_NO_FIRE))
```
Comment (Sirryan2002) [inline code/modules/mob/mob.dm:1032]:
```suggestion
					if(SS.cpu_display == SS_CPUDISPLAY_HIGH && !(SS.flags & SS_NO_FIRE))
```

File: code/controllers/subsystem/tickets/mentor_tickets.dm (hunk 3 above)
Comment (AffectedArc07) [reply code/controllers/subsystem/tickets/mentor_tickets.dm:24]: These should call parent a-la `atom/Initialize`.

Even if we override it, child SSs should do what their parents do.

File: code/__DEFINES/preferences.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/__DEFINES/preferences.dm:62]: Yes

File: code/datums/alarm_manager.dm (hunk 4 above)
Comment (AffectedArc07) [reply code/datums/alarm_manager.dm:12]: No

File: code/modules/mob/mob.dm (hunk 6 above)
Comment (AffectedArc07) [reply code/modules/mob/mob.dm:981]: I prefer having it like this and its easier to keep track, plus if this went at the start of the `if()` without the brackets they have issues.

I can change it if you really want but I would rather not.

File: code/modules/mob/mob.dm (hunk 7 above)
Comment (AffectedArc07) [reply code/modules/mob/mob.dm:1032]: Same here, just a styling thing and easier to keep track. The extra brackets have no perf implications.

File: code/controllers/subsystem/tickets/mentor_tickets.dm (hunk 3 above)
Comment (Sirryan2002) [reply code/controllers/subsystem/tickets/mentor_tickets.dm:24]: Makes sense 👍

File: code/modules/mob/mob.dm (hunk 6 above)
Comment (Sirryan2002) [reply code/modules/mob/mob.dm:981]: Feel free to leave it as is :D

**Review by Fox-McCloud — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]
looks good, LGTM

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**